Create a basic list of tools and configuration for developers

alexchapar opened the pull request:
Draft for some tooling that can be useful for devevelopers. It's essentially intended for developers on Windows ( but not limited to them 😉)

It includes configuration for Git, some useful registry, scripts to manage services, scripts and helper to work with Scoop.sh and also managing the extensions of VS Code

alexchapar committed:
Draft for some tooling that can be useful for dev

alexchapar committed:
draft for some tooling that can be usefull for dev

alexchapar committed:
Merge branch 'feature/create-basic-tooling-for-developers' of https://github.com/brave-simpletons/tooling-for-developers into feature/create-basic-tooling-for-developers

alexchapar committed:
fix usage for git delete-squashed

Bibz87-CG commented on README.md:
```diff
@@ -1 +1,21 @@
+
+Then execute the scripts. Each folder should contains a readme that explains how to use it.
+
+By the way, We know, it's not elegant but we do trust you will contribute with some PR to improve it :P
```
Trust your tools! :) Maybe update the sentence to something more positive? Something like "We're always improving and your contributions are welcome!"?

alexchapar replied:
I updated with your suggestions and changed some other stuff in the getting started. I also did some little changes to the blockquote

Bibz87-CG commented on git/README.md:
```diff
@@ -0,0 +1,19 @@
+
+This folder contains configuration for git and also the shell integration of the git-bash (using the `reg` files)
+
+These files are configured with the assumption that you are using [SCOOP](https://scoop.sh), as we do ;)
```
"as we do" isn't necessary. :)

ghost commented:
Why are the registry files binary?

ghost commented:
Missing VSCode Insiders export script?

alexchapar committed:
change the root readme based on some comments

alexchapar committed:
rename scripts and update the readme to manage services

alexchapar committed:
Remove a statement in git readme

alexchapar committed:
update the extensions' export/import for vscode

alexchapar commented:
> Missing VSCode Insiders export script?

I did change the way scripts for the extensions' export/import for VSCode* ia done... It now includes both. It's kinda "rigid" for now but I think it could be a good start

alexchapar committed:
remove backup folder in scoop scripts & gitignore

alexchapar committed:
add some color for vscode

alexchapar committed:
update filename for export/import for scoop

alexchapar committed:
Add extensions recommendations for contributors (and configs)

alexchapar committed:
Add badges in the readme at root

alexchapar committed:
Add contributor badge

alexchapar committed:
Move contributor badge

alexchapar commented:
@Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄

Bibz87-CG reviewed (changes requested):
> @Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄

I'm still seeing unresolved comments. 😛

alexchapar commented on README.md:
```diff
@@ -1 +1,21 @@
+
+You can contribute too, with your scripts or other nice "tooling for developers", using PR.
+
+Also do not forget to always tests your stuff on your end :P
```
As mentionned previously... 
> ... I [...] did some testing on my end. It all seems to be good... for now 😄

alexchapar replied:
unless you found something during your tests ?

alexchapar committed:
Update the git's readme based on comments

alexchapar committed:
Fix typos

alexchapar commented on vscode/vscode.import.extensions.ps1:
```diff
@@ -0,0 +1,16 @@
+    )
+
+    # import Extensions for VSCode
+    Get-Content "$Path\backup\vscode.extensions.txt" | ForEach-Object { code --install-extension $_ }
```
hmmm... would you suggest a note to add code in `PATH` ? cause trying to find out where the "code" is installed could be tough if not executed within it... ;)

alexchapar committed:
Swap function name

alexchapar committed:
Add a warning for code to be in PATH

alexchapar committed:
Minimal fixes

alexchapar commented:
> Why are the registry files binary?

I don't know... since we can view the file... it seems to not be shown in the `files changed` but we can view them (using the dots)

alexchapar commented:
> > @Bibz87-CG, I fixed all your comments, added some stuff and did some testing on my end. It all seems to be good... for now 😄
> 
> I'm still seeing unresolved comments. 😛

I did some changes... should be good now... 😄

alexchapar commented on vscode/README.md:
```diff
@@ -0,0 +1,35 @@
+
+## How to use it
+
+> [!WARNING]
```
This ia an excellent idea ;)

alexchapar replied:
It was an excellent idea
I used the emojis based on that suggestion : https://gist.github.com/rxaviers/7360908#gistcomment-3398183

alexchapar committed:
adding emoji for warning and note

alexchapar committed:
change/add emoji

alexchapar committed:
add overview section title (like other pages)

alexchapar committed:
remove toc

alexchapar commented on git/README.md:
```diff
@@ -0,0 +1,21 @@
+# Configuration for Git
+
+[[_TOC_]]
```
I removed the [[_TOC_]] since our readme are still little... ;)

alexchapar commented on scoop/README.md:
```diff
@@ -0,0 +1,46 @@
+# SCOOP
+
+[[_TOC_]]
```
I removed the [[_TOC_]] since our readme are still little... ;)

Bibz87-CG reviewed (approved):
All good! :)
